File: app/src/main/java/com/example/counter/SetupActivity.java
package com.example.counter;

import static android.view.View.*;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.ProgressBar;
import android.widget.Switch;

import androidx.appcompat.app.AppCompatActivity;

// SetupActivity.java
public class SetupActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setup);

        SharedPreferences preferences = getSharedPreferences("com.example.counter", Context.MODE_PRIVATE);
        if (!preferences.getBoolean("firstRun", true)) {
            // If it's not the first run, start MainActivity and finish SetupActivity
            startActivity(new Intent(SetupActivity.this, MainActivity.class));
            finishAffinity();
            return;
        }

        setContentView(R.layout.activity_setup);

        // Example: Button to finish setup and go to MainActivity
        @SuppressLint({"MissingInflatedId", "LocalSuppress"})

        Button finishSetup = findViewById(R.id.finishbtn);
        ProgressBar progressBar = findViewById(R.id.progressBar);

        @SuppressLint("UseSwitchCompatOrMaterialCode")
        Switch switch1 = findViewById(R.id.switch1);

        @SuppressLint("UseSwitchCompatOrMaterialCode")
        Switch switch2 = findViewById(R.id.switch2);


        preferences = getSharedPreferences("com.example.counter", Context.MODE_PRIVATE);
        switch1.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                // Save the state of the switch
                SharedPreferences.Editor editor = preferences.edit();
                editor.putBoolean("switch1", isChecked);
                editor.apply();
            }

        finishSetup.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                progressBar.setProgress(100);

                // Save first run preference
                SharedPreferences.Editor editor = preferences.edit();
                editor.putBoolean("firstRun", false);
                editor.apply();

                Intent intent = new Intent(SetupActivity.this, MainActivity.class);
                startActivity(intent);
                finishAffinity(); // Finish SetupActivity so it's removed from the back stack
            }
        });
    }
}

File: app/src/main/java/com/example/counter/SetupActivity.kt
package com.example.counter

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.CompoundButton
import android.widget.ProgressBar
import android.widget.Switch
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate

class SetupActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setup)
        var preferences = getSharedPreferences("com.example.counter", MODE_PRIVATE)
        if (!preferences.getBoolean("firstRun", true)) {
            startActivity(Intent(this@SetupActivity, MainActivity::class.java))
            finishAffinity()
            return
        }
        @SuppressLint("MissingInflatedId", "LocalSuppress") val finishbtn =
            findViewById<Button>(R.id.finishbtn)
        val progressBar = findViewById<ProgressBar>(R.id.progressBar)
        @SuppressLint("UseSwitchCompatOrMaterialCode") val switch1 =
            findViewById<Switch>(R.id.switch1)
        @SuppressLint("UseSwitchCompatOrMaterialCode") val switch2 =
            findViewById<Switch>(R.id.switch2)
        preferences = getSharedPreferences("com.example.counter", MODE_PRIVATE)
        val finalPreferences1 = preferences
        switch1.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            val editor = finalPreferences1.edit()
            editor.putBoolean("switch1", isChecked)
            editor.apply()
        }
        val isDarkModeOn = preferences.getBoolean("darkMode", false)
        switch2.setChecked(isDarkModeOn)
        val finalPreferences2 = preferences
        switch2.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            val editor = finalPreferences2.edit()
            editor.putBoolean("darkMode", isChecked)
            editor.apply()
            if (isChecked) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            } else {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            }
        }
        val finalPreferences = preferences
        finishbtn.setOnClickListener { v: View? ->
            progressBar.progress = 100
            val editor = finalPreferences.edit()
            editor.putBoolean("firstRun", false)
            editor.apply()
            val intent = Intent(this@SetupActivity, MainActivity::class.java)
            startActivity(intent)
            finishAffinity()
        }
    }
}